### ISSUE TITLE
Vanity URLs to folders do not work in 3.1

### ISSUE THREAD

**Issue by brentgriffin** [dotCMS/core#6912]
Automated test for test case 384 fails on 3.1 nightly builds.  This tests add a vanity URL that is supposed to redirect to "products/".  Instead of redirecting to this location, server returns a 404.

1.  Create a vanity URL for demo.dotcms.com that is to redirect to "products/"
![image](https://cloud.githubusercontent.com/assets/3208956/5653220/e1f85e22-967d-11e4-8afc-35630848e0ec.png)
2.  Access vanity URL (i.e. "http://localhost:8080/384") and receive 404 error page

This test fails on both 3.1 nightly builds but has run correctly on 3.0 builds many times.

Ubuntu 14.04 // Java (build 1.7.0_72-b14) // 3.1 Nightly build 2015-01-06 20:52 // H2 // Firefox & Chrome

**Comment by brentgriffin** [comment]
FYI - Test 386 also failed which is basically the same test except that it creates a vanity URL for "All Sites" instead of for "demo.dotcms.com"

**Comment by dsilvam** [comment]
Worth mentioning that the Redirect URL should be '/products/' instead of 'products/'. At least for 3.x

**Comment by brentgriffin** [comment]
If this new syntax is required, this is a problem because previously  "products/" for the URL to redirect to was permitted and works.  If this is changed, could result in broken customer Virtual URLS

**Comment by wezell** [comment]
Not sure this is a good fix.  I am getting a null pointer if I hit the home page with url "/":

```
Jan 10, 2015 9:53:01 AM org.apache.catalina.core.StandardWrapperValve invoke
SEVERE: Servlet.service() for servlet [jsp] in context with path [] threw exception
java.lang.NullPointerException
	at com.dotmarketing.filters.CMSFilter.doFilter(CMSFilter.java:191)
```

The reason is that the
the CmsUrlUtil.isVanityUrl() is returning false for the url "/" because line 134 

https://github.com/dotCMS/core/blob/7c60b03cb32484fb9a6e514078f59665bfceb5eb/src/com/dotmarketing/filters/CmsUrlUtil.java#L134

This strips the  "/" and gives us a url of "" and then line 142 doesn't work.

**Comment by wezell** [comment]
Looking at this, it seems like the cure is worse than the disease.  I think @dsilvam  is on to something. I would vote to make "products/" an invalid vanity url - as a relative path, it really shouldn't work by any convention I can think of and is more a function of the old and crappy edit a vanity url logic.  Let's talk Monday?

**Comment by jtesser** [comment]
We can talk but I do not think we should break all people who have had this. We have tests that worked prior and now fail. We did this on purpose FWIW

**Comment by jtesser** [comment]
@wezell NPE issue is addressed now FWIW

**Comment by bryanboza** [comment]
Fixed, tested on master-3.1 // H2 // FF
